tutorials/foundry-start-to-end.md

```diff
@@ -0,0 +1,365 @@
+
+    uint256 constant CAPACITY = 100;
+
+    function setUp() public {
+        token = new MyToken(1000);
+        container = new Container(token, CAPACITY);
+    }
+
+    function testInitialUnsatisfied() public {
+        assertEq(token.balanceOf(address(container)), 0);
+        assertTrue(container.status() == ContainerStatus.Unsatisfied);
+    }
+
+    function testContainerFull() public {
+        token.transfer(address(container), CAPACITY);
+        container.updateStatus();
+
+        assertEq(token.balanceOf(address(container)), CAPACITY);
+        assertTrue(container.status() == ContainerStatus.Full);
+    }
+}
+```
+
+This test smart contract has two tests, so when running the tests, there will be two deployments of both `MyToken` and `Container`, for four smart contracts in total.  
+
+### Test Harnesses in Foundry {: #test-harnesses-in-foundry }
+
+Sometimes you'll want to test an `internal` function in a smart contract. To do so, you'll have to write a test harness smart contract, which inherits from the smart contract and exposes the internal function as a public one.  
+
+For example, in `Container`, there is an internal function named `_isOverflowing`, which checks to see if the smart contract has more tokens than its capacity. To test this, add the following test harness smart contract to the `Container.t.sol` file:  
+
+```solidity
+contract ContainerHarness is Container {
+    constructor(MyToken _token, uint256 _capacity) Container(_token, _capacity) {}
+
+    function exposed_isOverflowing(uint256 balance) external view returns(bool) {
+        return _isOverflowing(balance);
+    }
+}
+```  
+
+Now, inside of the `ContainerTest` smart contract, you can add a new test that tests the previously unreachable `_isOverflowing` contract:  
+
+```solidity
+    function testIsOverflowingFalse() public {
+        ContainerHarness harness = new ContainerHarness(token , CAPACITY);
+        assertFalse(harness.exposed_isOverflowing(CAPACITY - 1));
+        assertFalse(harness.exposed_isOverflowing(CAPACITY));
+        assertFalse(harness.exposed_isOverflowing(0));
+    }
+```
+
+Now, when you run the test with `forge test`, you should see the following:  
+
+**PUT THE CONSOLE IMAGE HERE**  
+
+### Fuzzing Tests in Foundry {: #fuzzing-tests-in-foundry}
+
+### Forking Tests in Foundry {: #forking-tests-in-foundry}
+
+## Deploying in Foundry with Solidity Scripts {: #deploying-in-foundry-with-solidity-scripts }
+
+Not only are tests in Foundry written in Solidity, the scripts are too! Like other developer environments, scripts can be written to help interact with deployed smart contracts or can help along a complex deployment process that would be difficult to do manually. Even though scripts are written in Solidity, they are never deployed to a chain. Instead, much of the logic is actually ran off-chain, so don't worry about any additional gas costs for using Foundry instead of a JavaScript environment like HardHat.  
```

Comment:
```suggestion
Not only are tests in Foundry written in Solidity, the scripts are too! Like other developer environments, scripts can be written to help interact with deployed smart contracts or can help along a complex deployment process that would be difficult to do manually. Even though scripts are written in Solidity, they are never deployed to a chain. Instead, much of the logic is actually run off-chain, so don't worry about any additional gas costs for using Foundry instead of a JavaScript environment like HardHat.  
```